CHANGELOG.md

```diff
@@ -178,6 +178,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Deprecated `Trainer.should_rank_save_checkpoint` Trainer property ([#11068](https://github.com/PyTorchLightning/pytorch-lightning/pull/11068))
 
+
+- Deprecated most of the callback_hook functions in `TrainerCallbackHookMixin` ([#11148](https://github.com/PyTorchLightning/pytorch-lightning/pull/11148))
```

Comment:
If the comment here holds: https://github.com/PyTorchLightning/pytorch-lightning/pull/11148#discussion_r771886148

We can later edit this to be Deprecated `TrainerCallbackHookMixin` so there's no ambiguity around which callback hooks are going to be removed